AndroidSDKFcm/src/main/java/com/leanplum/LeanplumPushFirebaseMessagingService.java

```diff
@@ -65,6 +71,13 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
         Leanplum.setApplicationContext(this);
       }
       Map<String, String> messageMap = remoteMessage.getData();
+      try {
+        if (isDuplicateNotification(messageMap)) {
```

Comment:
We have existing method LPActionManager in LPActionManager.m, which needs to be addressed as well,  otherwise we will have 2 duplicate clearing methods. Which will be wrong.

Also, i would not recommend checking for duplicates on "onMessageRecieved", rather implement this logic in LPActionManager, where the existing dedup implementation is there. This does not gurantee that the previous message was displayed.